pkgs/by-name/wr/wrangler/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  pkgs,
+  stdenv,
+  fetchFromGitHub,
+  pnpm_9,
+  nodejs,
+  npmHooks,
+  writeScriptBin
+}:
+let
+  srcHash = "sha256-/4iIkvSn85fkRggmIha2kRlW0MEwvzy0ZAmIb8+LpZQ=";
+  pnpmDepsHash = "sha256-aTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=";
+in
+  stdenv.mkDerivation (finalAttrs: {
+  pname = "wrangler";
+  version = "3.62.0";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = "workers-sdk";
+    rev = "wrangler@${finalAttrs.version}";
+    hash = "${srcHash}";
+  };
+
+  buildInputs = [
+    pkgs.llvmPackages.libcxx
+    pkgs.llvmPackages.libunwind
+    pkgs.musl
+    pkgs.xorg.libX11
+  ] ++ lib.optional stdenv.isLinux pkgs.autoPatchelfHook;
+
+  nativeBuildInputs = [
+    nodejs
+    pnpm_9.configHook
+    pkgs.autoPatchelfHook
+  ];
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "${pnpmDepsHash}";
+  };
+
+  preBuild = ''
+    addAutoPatchelfSearchPath lib/node_modules/.pnpm/@cloudflare+workerd-linux-64@1.20240620.1/node_modules/@cloudflare/workerd-linux-64/bin/
+    '';
+
+  # @cloudflare/vitest-pool-workers wanted to run a server as part of the build process
+  # so I simply removed it
+  postBuild = ''
+    rm -fr packages/vitest-pool-workers
+    NODE_ENV="production" pnpm --filter miniflare run build
+    NODE_ENV="production" pnpm --filter wrangler run build
+    # pnpm --offline deploy --frozen-lockfile  --ignore-script  --filter=bash-language-server server-deploy
+    '';
+
+  # this was taken from a previous script which was generated somehow
```

Comment:
Could you expand on that a little bit? Generated where?

Reply:
I copied this from a previous version of `wrangler` somehow but now I can't find it; at the time, the point was to see if we could get it working at all, so it was a quick and dirty way to move forward.

Having looked a little more carefully, I think it could be possible to use the `wrangler.js` file directly which is put in `bin` (ie renaming it to `wrangler`)...I can try this out @dezren39 and it could make things a bit cleaner.

Reply:
I see this has been completely changed so I guess there is nothing to do here - thantks @dezren39!